File: 3_private_encaptulaiton.py
class MyClass:
    def __init__(self) -> None:
        self.__private_var = 42 # private variable

    def __private_method(self):  # private method
        print("This is a private method")

    def public_method(self):
        print("Accessing private method: ",self.__private_var)
        self.__private_method() # accessing private method within the class


obj = MyClass()
obj.public_method()

#trying to access private members from outside the class

try:
    print(obj.__private_var)
except AttributeError:
    print("Cannot access __private_var outside the class")

try:
    obj.__private_method()
except AttributeError:
    print("Cannot access __private_method outside the class")

File: 3_private_encaptulaiton.cpp
#include <bits/stdc++.h>
using namespace std;

class MyClass{
    private:
    int private_var;//private variable

    void private_method(){ //private method
        cout<<"This is a private method"<<endl;
    }

    public:
    MyClass(){
        private_var = 42;
    }
    void public_method(){
        cout<<"Accessing private variable: "<< private_var <<endl;
        private_method(); // Accessing private method within the class
    }
};

int main()
{
    MyClass obj;
    obj.public_method();
    // Trying to access private members from outside the class
    // Uncommenting the lines below will cause a compile-time error
    // cout << obj.private_var;
    // obj.private_method();
    return 0;
}